### PR TITLE
Improve event types

### PR DESCRIPTION
**Pull request by alexgleason**
Fixes typescript issues when trying to pass events of unknown kind into functions that shouldn't care. Functions that don't care about kind should accept an `Event<number>`.

More comments below.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -103,7 +103,6 @@
     "no-octal-escape": 2,
     "no-path-concat": 0,
     "no-proto": 2,
-    "no-redeclare": 2,
```
Comment (alexgleason) [inline .eslintrc.json:106]: I had to remove this to make TypeScript function overloading work. We could enable the TypeScript version if we want, but I didn't see the point. I think TS will warn you anyway.

File: event.ts [hunk 2]
```diff
@@ -20,6 +19,7 @@ export enum Kind {
   ChannelMessage = 42,
   ChannelHideMessage = 43,
   ChannelMuteUser = 44,
+  Blank = 255,
```
Comment (alexgleason) [inline event.ts:22]: I changed the name of this from #200. The other name was overly verbose.

File: nip57.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export function useFetchImplementation(fetchImplementation: any) {
   _fetch = fetchImplementation
 }
 
-export async function getZapEndpoint(metadata: Event): Promise<null | string> {
+export async function getZapEndpoint(metadata: Event<0>): Promise<null | string> {
```
Comment (alexgleason) [inline nip57.ts:16]: This is the only potentially breaking change. TypeScript users will have to update their code to: `getZapEndpoint(event as Event<0>)`.

Once we update relay.ts to infer types from filters (in a later MR) this shouldn't be necessary.

File: .eslintrc.json (hunk 1 above)
Comment (fiatjaf) [reply .eslintrc.json:106]: Yes, please. eslint is stupid.

**Pull request merged**